core/src/main/java/tech/tablesaw/io/csv/CsvWriter.java

```diff
@@ -61,7 +61,7 @@ public void write(Table table, CsvWriteOptions options) {
     } finally {
       if (csvWriter != null) {
         csvWriter.flush();
-        csvWriter.close();
+        if (options.autoClose()) csvWriter.close();
```

Comment:
`FixedWidthWriter` would need to be updated as well

Reply:
> `FixedWidthWriter` would need to be updated as well

Fixed. But I have some problem with `FixedWidthWriter` when I try to output the table. 

```java
ByteArrayOutputStream baos = new ByteArrayOutputStream();
FixedWidthWriteOptions options =
    new FixedWidthWriteOptions.Builder(baos)
        .header(new FixedWidthFields(10,10))
        .build();
FixedWidthWriter writer = new FixedWidthWriter();
writer.write(table, options);
```

The last line of code got IllegalStateException. When `FixedWidthWriterSettings` is constructed with `FixedWidthWriteOptions`, `fieldLengths` is neved assigned. I think it will be fixed in another issue.